### PR TITLE
Linked list insertions

### PR DESCRIPTION
**Pull request by L-nobilis**
- [x] Top-level README “Table of Contents” is updated
 - [x] README for this challenge is complete
       - [x] Summary, Description, Approach & Efficiency, Solution
       - [ ] Picture of whiteboard
       - [x] Link to code
 - [x] Feature tasks for this challenge are completed
 - [x] Unit tests written and passing
       - [x] “Happy Path” - Expected outcome
       - [ ] Expected failure
       - [ ] Edge Case (if applicable/obvious)

### CODE REVIEW

File: python/data_structures/linked_list.py [hunk 1]
```diff
@@ -36,10 +36,46 @@ def includes(self, target_value):
...
+    def insert_before(self, target_value, inserted_value):
+        """
+        Adds a new node with the given new value immediately before the first node that has the value specified. If target_value is not found, it does nothing.
+        """
+        current = self.head
+
+        while current:
+            if current.value == target_value:
+                self.insert(inserted_value)
+                break
```
Comment (jstreifel-33) [inline python/data_structures/linked_list.py:60]: I saw you have a note about moving this `if` statement outside the while loop. Definitely would recommend it, since this condition is only really applicable when we're looking at the head node.
